backend/workflow_manager/endpoint/database_utils.py

```diff
@@ -284,22 +299,76 @@ def execute_write_query(
         - Snowflake does not support INSERT INTO ... VALUES ...
           syntax when VARIANT columns are present (JSON).
           So we need to use INSERT INTO ... SELECT ... syntax
+        - sql values can contain data with single quote. It needs to
         """
-        sql = (
-            f"INSERT INTO {table_name} ({','.join(sql_keys)}) "
-            f"SELECT {','.join(sql_values)}"
-        )
-        logger.debug(f"insertng into table with: {sql} query")
+        sql = DBConnectorQueryHelper.build_sql_insert_query(table_name, sql_keys)
+        logger.debug(f"inserting into table {table_name} with: {sql} query")
+
+        sql_values = DBConnectorQueryHelper.prepare_sql_values(cls_name, sql_values)
+        logger.debug(f"sql_values: {sql_values}")
+
         try:
-            if hasattr(engine, "cursor"):
-                with engine.cursor() as cursor:
-                    cursor.execute(sql)
-                engine.commit()
+            if cls_name == DBConnectionClass.BIGQUERY:
+                DBConnectorQueryHelper.execute_bigquery(
+                    engine, table_name, sql_keys, sql_values
+                )
+            elif hasattr(engine, "cursor"):
+                DBConnectorQueryHelper.execute_cursor_query(engine, sql, sql_values)
             else:
-                engine.query(sql)
-        except Exception as e:
-            logger.error(f"Error while writing data: {str(e)}")
-            raise e
+                DBConnectorQueryHelper.execute_generic_query(engine, sql, sql_values)
+        except PsycopgError.UndefinedTable as e:
```

Comment:
@chandrasekharan-zipstack Do we need all these errors to be captured separately ? 
If YES, @kirtimanmishrazipstack lets refactor this a bit by creating a static method to capture these errors and log and raise exception from there.

Reply:
@athul-rs same reason refactoring and moving to base unstract_db class is not helpful as mentioned here https://github.com/Zipstack/unstract/pull/330#discussion_r1617174048